primitives/arithmetic/src/fixed.rs

```diff
@@ -0,0 +1,1527 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Decimal Fixed Point implementations for Substrate runtime.
+
+use sp_std::{ops::{self, Add, Sub, Mul, Div}, fmt::Debug, prelude::*, convert::{TryInto, TryFrom}};
+use codec::{Encode, Decode};
+use crate::{
+	helpers_128bit::multiply_by_rational, PerThing,
+	traits::{
+		SaturatedConversion, CheckedSub, CheckedAdd, CheckedMul, CheckedDiv, CheckedNeg,
+		Bounded, Saturating, UniqueSaturatedInto, Zero, One, Signed
+	},
+};
+
+#[cfg(feature = "std")]
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+/// Integer types that can be used to interact with `FixedPointNumber` implementations.
+pub trait FixedPointOperand: Copy + Clone + Bounded + Zero + Saturating
+	+ PartialOrd + UniqueSaturatedInto<u128> + TryFrom<u128> + CheckedNeg {}
+
+impl FixedPointOperand for i128 {}
+impl FixedPointOperand for u128 {}
+impl FixedPointOperand for i64 {}
+impl FixedPointOperand for u64 {}
+impl FixedPointOperand for i32 {}
+impl FixedPointOperand for u32 {}
+impl FixedPointOperand for i16 {}
+impl FixedPointOperand for u16 {}
+impl FixedPointOperand for i8 {}
+impl FixedPointOperand for u8 {}
+
+/// Something that implements a fixed point with an arbitrary precision `DIV`,
+/// i.e. `1 / DIV` can be represented. `DIV` should be a power of `10`.
```

Comment:
```suggestion
/// i.e. `1 / DIV` can be represented. `DIV` should be a power of `10`.
/// 
/// Each type will a theoretical limit that it can store. This is equal to `Self::Inner::max_value() / Self::DIV`.
```